src/Solvers/krylov.jl

```diff
@@ -0,0 +1,152 @@
+import Krylov
+import Krylov.FloatOrComplex
+
+using KernelAbstractions: @index, @kernel
+
+struct KrylovField{T, F <: AbstractField} <: AbstractVector{T}
+    field::F
+end
+
+function KrylovField(field::F) where F <: AbstractField
+    T = eltype(field)
+    return KrylovField{T,F}(field)
+end
+
+Base.size(kf::KrylovField) = size(kf.field)
+Base.length(kf::KrylovField) = length(kf.field)
+Base.getindex(kf::KrylovField, i::Int) = getindex(kf.field, i)
+
+function Krylov.kdot(n::Int, x::KrylovField, y::KrylovField)
+    mx, nx, kx = size(x.field)
+    _x = x.field
+    _y = y.field
+    T = eltype(kf.field)
+    res = zero(T)
+    for i = 1:mx
+        for j = 1:nx
+            for k = 1:kx
+                res += _x[i,j,k] * _y[i,j,k]
+            end
+        end
+    end
+    return res
+end
+
+function Krylov.knorm(n::Int, x::KrylovField)
+    mx, nx, kx = size(x.field)
+    _x = x.field
+    T = eltype(kf.field)
+    res = zero(T)
+    for i = 1:mx
+        for j = 1:nx
+            for k = 1:kx
+                res += _x[i,j,k]^2
+            end
+        end
+    end
+    return sqrt(res)
+end
+
+function Krylov.kscal!(n::Int, s, x::KrylovField)
+    mx, nx, kx = size(x.field)
+    _x = x.field
+    for i = 1:mx
+        for j = 1:nx
+            for k = 1:kx
+                _x[i,j,k] = s * _x[i,j,k]
+            end
+        end
+    end
+    return x
+end
```

Comment:
```suggestion
function Krylov.kscal!(n::Int, s, x::KrylovField)
    xp = parent(x.field)
    xp .*= s
    return x
end
```

will this be used by the solver?

Reply:
Yes, but not all Krylov solvers.